worker/raft/raftforwarder/metrics.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package raftforwarder
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	metricsNamespace = "juju_raftforwarder"
+)
+
+// metricsCollector is a prometheus.Collector that collects metrics
+// about how long it's taking to forward requests to raft and get
+// responses.
+type metricsCollector struct {
+	requests *prometheus.SummaryVec
+}
+
+func newMetricsCollector() *metricsCollector {
+	return &metricsCollector{
+		requests: prometheus.NewSummaryVec(prometheus.SummaryOpts{
+			Namespace: metricsNamespace,
+			Name:      "request",
+			Help:      "Request times for raft forwarder operations in ms",
+			Objectives: map[float64]float64{
+				0.5:  0.05,
```

Comment:
I don't understand this map.

Also, what is the difference between a SummaryVec and a GuageVec?

Reply:
Summary is what produces the quantiles we use in the raft charts. Gauges are more limited - you can't calculate a rate for them, for example. We use them to track number of models, the docs use them for things like temperature - quantities that go up and down but don't represent specific events.
https://prometheus.io/docs/concepts/metric_types/

The map sets what quantiles the summary collects and the estimation error for each quantile, as far as I understand it. I copied those values from the docs - they're the defaults but they say not to rely on the defaults since they'll be changing at some point in the future. 
https://godoc.org/github.com/prometheus/client_golang/prometheus#Summary
https://godoc.org/github.com/prometheus/client_golang/prometheus#SummaryOpts

Reply:
OK